### PR TITLE
add --destination-use-user-folder option to files:transfer-ownership command

### PR DESCRIPTION
**Pull request by mrow4a**

### CODE REVIEW

**Comment by mmattel** [comment]
@mrow4a you need to add the new option also to the corresponding occ command set file

**Comment by mrow4a** [comment]
@mmattel done, but not sure if correctly

File: modules/admin_manual/pages/configuration/server/occ_commands/core_commands/file_commands.adoc [hunk 1]
```diff
@@ -265,13 +265,19 @@ This command is useful before removing users.
...
+| `--destination-use-user-folder` | if destination user never logged in + 
+and thus has no mounts transfer + 
+directly to top-level user folder.
 |===
 
-For example, to move all files from `<source-user>` to `<destination-user>`, use the following command:
+
+
+
+For example, to move all files from `<source-user>` to `<destination-user>` transfer folder, use the following command:
```
Comment (mmattel) [inline modules/admin_manual/pages/configuration/server/occ_commands/core_commands/file_commands.adoc:280]:
```suggestion

For example, to move all files from `<source-user>` to `<destination-user>` transfer folder, use the following command:
```

File: modules/admin_manual/pages/configuration/server/occ_commands/core_commands/file_commands.adoc [hunk 2]
```diff
@@ -291,6 +295,17 @@ Ownership of `folder/to/move` and all files and folders which it contains will b
     <destination-user>
 ----
 
+In case when entire user folder of `<source-user>` needs to be migrated to `<destination-user>` user folder, use the `--destination-use-user-folder` switch, as in the example below.
```
Comment (phil-davis) [inline modules/admin_manual/pages/configuration/server/occ_commands/core_commands/file_commands.adoc:298]:
```suggestion
If the entire user folder of `<source-user>` needs to be migrated to `<destination-user>` user folder, use the `--destination-use-user-folder` switch, as in the example below.
```

File: modules/admin_manual/pages/configuration/server/occ_commands/core_commands/file_commands.adoc [hunk 3]
```diff
@@ -291,6 +295,17 @@ Ownership of `folder/to/move` and all files and folders which it contains will b
...
+Destination user needs to be created but never log in, we recommend using owncloud maintenance mode for this operation.
```
Comment (phil-davis) [inline modules/admin_manual/pages/configuration/server/occ_commands/core_commands/file_commands.adoc:299]:
```suggestion
The destination user needs to be created but never log in, we recommend using owncloud maintenance mode for this operation.
```

**Comment by phil-davis** [comment]
Squashed and rebased.

**Comment by phil-davis** [comment]
This is a new thing that will be in the next oC10.9 release. So no backport needed.

**Review by phil-davis — APPROVED** [review]

File: modules/admin_manual/pages/configuration/server/occ_commands/core_commands/file_commands.adoc [hunk 4]
```diff
@@ -265,13 +265,17 @@ This command is useful before removing users.
...
 |===
-| `--path=[PATH]`        | selectively provide the path to transfer. +
+| `--path=[PATH]`                 | selectively provide the path to transfer. +
 For example --path="folder_name"
 | `-s, +
---accept-skipped-shares` | always confirm to continue in case of skipped shares.
+--accept-skipped-shares`          | always confirm to continue in case of skipped shares.
+| `--destination-use-user-folder` | if destination user never logged in + 
+and thus has no mounts transfer + 
+directly to top-level user folder.
```
Comment (mmattel) [inline modules/admin_manual/pages/configuration/server/occ_commands/core_commands/file_commands.adoc:274]: The first line is a if
followed by a statemet - ok
but the next line does not point to a solution or explanation (directly to top-level user folder).
The sentence looks "strange/confusing"

Well done, your changes render the following ✔️ 

image

Comment (mrow4a) [reply modules/admin_manual/pages/configuration/server/occ_commands/core_commands/file_commands.adoc:274]: any suggestions how to improve? my docs were really engineer driven

Comment (EParzefall) [reply modules/admin_manual/pages/configuration/server/occ_commands/core_commands/file_commands.adoc:274]: Yes, this is difficult to parse. A comma after "mounts" should clarify things. Then the share is transferred directly to the top-level user folder.

Comment (mrow4a) [reply modules/admin_manual/pages/configuration/server/occ_commands/core_commands/file_commands.adoc]: @mmattel @EParzefall adjusted, can you check now ?

**Comment by mmattel** [comment]
@mrow4a thansk for the update. 

Q: `transfer directly to top-level user folder if destination user never logged in.`

Does this mean the following:

If the target user never logged in, there is no skeleton on a directory level present for this user.
when using this option, it checks about this fact, creates the skeleton and moves the source directly in the root of the users home - correct?

What happens if this option is not set and the user never logged in? 
Is there an error/warning message pointing to use that option?

**Comment by mrow4a** [comment]
@mmattel updated

**Review by mmattel — APPROVED** [review]

**Comment by mmattel** [comment]
@mrow4a thanks for updating - approved.

No backport needed, will go into 10.9

**Pull request merged**